uploadservice-s3/src/main/java/net/gotev/uploadservice/s3/S3ClientWrapper.kt

```diff
@@ -0,0 +1,80 @@
+package net.gotev.uploadservice.s3
+
+import android.content.Context
+import android.content.Intent
+import com.amazonaws.auth.CognitoCachingCredentialsProvider
+import com.amazonaws.mobileconnectors.s3.transferutility.*
+import com.amazonaws.regions.Region
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.s3.AmazonS3Client
+import com.amazonaws.services.s3.model.CannedAccessControlList
+import net.gotev.uploadservice.data.UploadFile
+import net.gotev.uploadservice.logger.UploadServiceLogger
+import java.io.Closeable
+import java.io.File
+import java.lang.Exception
+
+class S3ClientWrapper (private val uploadId: String,
+                       observer: Observer,
+                       context: Context,
+                       identityPoolId: String,
+                       region: Regions) : Closeable {
+
+    private val credentialsProvider = CognitoCachingCredentialsProvider(context, identityPoolId, region)
+    private val s3: AmazonS3Client = AmazonS3Client(credentialsProvider, Region.getRegion(region))
+    private val transferUtility = TransferUtility.builder().s3Client(s3).context(context).build()
+    private lateinit var transferObserver : TransferObserver
+    private lateinit var uploadingFile: UploadFile
+
+    init {
+        context.startService(Intent(context, TransferService::class.java))
+    }
+
+    interface Observer {
+        fun onStateChanged(client: S3ClientWrapper, uploadFile: UploadFile, id: Int, state: TransferState?)
+        fun onProgressChanged(client: S3ClientWrapper, id: Int, bytesCurrent: Long, bytesTotal: Long)
+        fun onError(client: S3ClientWrapper, id: Int, ex: Exception?)
+    }
+
+    private val transferListener = object : TransferListener {
+        override fun onStateChanged(id: Int, state: TransferState?) {
+            observer.onStateChanged(this@S3ClientWrapper, uploadingFile, id, state)
+        }
+
+        override fun onProgressChanged(id: Int, bytesCurrent: Long, bytesTotal: Long) {
+            observer.onProgressChanged(this@S3ClientWrapper, id, bytesCurrent, bytesTotal)
+        }
+
+        override fun onError(id: Int, ex: Exception?) {
+            observer.onError(this@S3ClientWrapper, id, ex)
+        }
+    }
+
+    @Throws(Exception::class)
+    fun uploadFile(
+            context: Context,
+            bucketName :String,
+            serverSubPath: String,
+            uploadFile: UploadFile) {
+        UploadServiceLogger.debug(javaClass.simpleName, uploadId) {
+            "Starting S3 upload of: ${uploadFile.handler.name(context)}"
+        }
+        uploadingFile = uploadFile
+        val file = File(uploadFile.path)
+        transferObserver = transferUtility.upload(
+                bucketName,
+                serverSubPath + "/" + file.name,
+                file,
+                CannedAccessControlList.Private
```

Comment:
shouldn't the default access be part of the params for the file or upload? we will never change this, but would end users?